SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java

```diff
@@ -169,16 +105,17 @@ public void tokenize_onCreateMandateRequestSuccess_whenMandateAlreadyApproved_on
                 .build();
 
         SEPADirectDebitClient sut =
-                new SEPADirectDebitClient(activity, lifecycle, braintreeClient, sepaDirectDebitApi);
-        sut.setListener(listener);
+                new SEPADirectDebitClient(braintreeClient, sepaDirectDebitApi);
 
-        sut.tokenize(activity, sepaDirectDebitRequest);
-        verify(listener).onSEPADirectDebitSuccess(nonce);
-        verify(braintreeClient).sendAnalyticsEvent("sepa-direct-debit.tokenize.success");
+        sut.tokenize(sepaDirectDebitRequest, (sepaDirectDebitResponse, error) -> {
+            assertEquals(sepaDirectDebitResponse.getNonce(), nonce);
```

Comment:
Just confirming - are these callback assertions being invoked? I haven't seen this done without a countdown latch before

Reply:
Yes, these are! (Confirmed with breakpoints & making sure the tests failed when modified)

However I pushed a commit (a1b0434bf5b3f4c123fed315201c3e2db2e1c31b) to move these tests away from the callback syntax and to instead use ArgumentCaptor, to match with the standard way in this test file.

Reply:
Oh cool we maybe should move to that pattern eventually - seems cleaner. But good to go with the current pattern for now for consistency 👍